dpe/src/context.rs

```diff
@@ -382,6 +383,19 @@ mod tests {
         assert!(iter.next().unwrap().is_ok());
     }
 
+    #[test]
+    fn test_child_to_root_iter_infinite_loop() {
+        let contexts = [CONTEXT_INITIALIZER; MAX_HANDLES];
+        let mut i = 0;
+        for _ in ChildToRootIter::new(30, &contexts) {
+            i += 1;
+            // fail test if we iterate over all nodes without terminating
+            if i > MAX_HANDLES {
+                assert!(false);
```

Comment:
nit: Prefer `panic!("Some useful message")`. This makes debugging test failures easier.

Reply:
Done